core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -980,11 +980,15 @@ public Builder addSortOrder(SortOrder order) {
     }
 
     public Builder addSnapshot(Snapshot snapshot) {
-      if (snapshot == null || snapshotsById.containsKey(snapshot.snapshotId())) {
+      if (snapshot == null) {
         // change is a noop
         return this;
       }
 
+      ValidationException.check(!snapshotsById.containsKey(snapshot.snapshotId()),
+          "Snapshot already exist for the id: %s",
+          snapshot.snapshotId());
```

Comment:
@RussellSpitzer, what do you think about this? Any situations where you think this would cause an issue because we add the same snapshot and rely on the operation being idempotent?